### PR TITLE
PhpDoc use string[][] for arrays of string arrays

### PR DESCRIPTION
**Pull request by Maks3w**

### CODE REVIEW

File: src/RequestInterface.php [hunk 1]
```diff
@@ -33,7 +33,7 @@
      *
      * @see MessageInterface::getHeaders()
      * @see UriInterface::getHost()
-     * @return array Returns an associative array of the message's headers. Each
+     * @return string[][] Returns an associative array of the message's headers. Each
```
Comment (GrahamCampbell) [inline src/RequestInterface.php]: Doesn't "Each" now need to go on the next line?

**Comment by weierophinney** [comment]
I've never seen that particular notation (I've seen `string[]` but not the additional nesting). Does phpdoc even support that?

**Comment by GrahamCampbell** [comment]
> Does phpdoc even support that?

Probably depends on who's implemented the parser. :)

**Comment by weierophinney** [comment]
@GrahamCampbell Not terribly helpful; perhaps I should put it another way: are there any known phpdoc parsers that will understand that notation?

**Comment by GrahamCampbell** [comment]
> are there any known phpdoc parsers that will understand that notation?

I think phpDocumentor supports it.

**Comment by GrahamCampbell** [comment]
The whole phpdoc deal is very unofficial anyway. I wish the docblock PSR would hurry up. TBH, we should probably check that to see if it allowed the nested array syntax.

**Comment by weierophinney** [comment]
I've pinged Mike van Riel on twitter to find out.

**Comment by weierophinney** [comment]
Mike indicates it *is* supported. @Maks3w - if you want it in here, make a PR against php-fig/fig-standards, as this repo is supposed to mirror that one.

**Comment by weierophinney** [comment]
@php-fig/secretaries :+1:; in fact, the change is already present in the PSR-7 document, and was from a PR @Maks3w made there, so merging this would update the repo to be in line with the specification.

**Comment by michaelcullum** [comment]
@weierophinney `getHeaders()` doesn't exist on this interface anymore?

**Comment by weierophinney** [comment]
@michaelcullum It's on `MessageInterface` (as headers are common to all types). Not sure why it was ever on `RequestInterface` in this repo...

**Comment by michaelcullum** [comment]
Fixed and merged in d0aabc038f1f3ff665dfe1e9673bcd0df609c681

**Pull request closed without merging**